runtime/gc_base/StringTable.cpp

```diff
@@ -623,6 +623,7 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 				while (tempLength != 0) {
 					U_16 unicode = 0;
 					UDATA consumed = VM_VMHelpers::decodeUTF8CharN(tempData, &unicode, tempLength);
+					Assert_GC_true_with_message2(MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread), 0 != consumed, "Invalid UTF8 character at %p, %lu", tempData, tempLength);
```

Comment:
We always use `%zu` for `UDATA` type instead of `%lu` is used. I believe it works better for `long long` type

Reply:
Yes, `z` for UDATA. Are those asserts smart enough to evaluate the assert condition before the other arguments?

Reply:
Yes it is. Also further handling is done in "prevent inline" style function in .cpp, so it should not increase stack frame size.

Reply:
OK let's wait for the `zu` change before merge.

Reply:
Fixed.